File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    private int hourToMinute(String s){
        int h = Integer.parseInt(s.substring(0,2)) * 60;
        int m = Integer.parseInt(s.substring(s.length() - 2));
        return h + m;
    }
    private String minuteToHour(int n){
        String h = "00" + n / 60;
        String m = "00" + n % 60;
        return h.substring(h.length() -2) + ":" + m.substring(m.length() - 2);
    }
    
    public String solution(String video_len1, String pos1, String op_start1, String op_end1, String[] commands) {
        int videoLen = hourToMinute(video_len1);
        int pos = hourToMinute(pos1);
        int opStart = hourToMinute(op_start1);
        int opEnd = hourToMinute(op_end1);
        int total = pos;
        if(opStart <= total && total <= opEnd) total = opEnd;
        
        for(String s : commands){
            int num = s.equals("next") ? 10 : -10;
            total += num;
            total = Math.max(0, total);
            total = Math.min(videoLen, total);
            if(opStart <= total && total <= opEnd) total = opEnd;
        }
        
        return minuteToHour(total);
    }
}
/*
영상 처음위치 : 0분0초
영상 마지막위치 : 동영상의 길이
오프닝 구간 자동 오프닝 끝 구간으로
1. 오프닝 구간인지 확인
2. 더하거나 빼서 위치 확인 
    2-1. 10초 미만이면 0분 0초로
    2-2. 비디오 길이 초과이면 자동 비디오길이로 이동
3. 계산한 시간 출력

*/

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def tos(time):
    time = time.split(":")
    return int(time[0])*60 + int(time[1])
def tom(ss):
    m = "00" + str(ss // 60)
    s = "00" + str(ss % 60)
    return m[-2:] + ":" + s[-2:]

def solution(video_len, pos, op_start, op_end, commands):
    video_len = tos(video_len)
    pos = tos(pos)
    op_start, op_end = tos(op_start), tos(op_end)
    if op_start <= pos <= op_end:
            pos = op_end
    for cmd in commands:
        if cmd == "next":
            pos = min(pos + 10,  video_len)
        else:
            pos = max(pos - 10, 0)
        if op_start <= pos <= op_end:
            pos = op_end
    return tom(pos)
    
"""
[조건]
10초 전, 10초 후, 오프닝 스킵
10초 전: "prev" 입력 => 10초전 이동, 10초 미만은 0분0초로 이동
10초 후: "next" 입력 => 10초후 이동, 10초 미만은 마지막으로 이동
오프닝스킵: 현재 오프닝구간일 경우 자동으로 op_end로 이동
입력이 모두 끝난 후 동영상의 위치

[아이디어]
commands를 돌면서 명령 수행, 모든 시간을 초단위로 변형후 계산
[자료구조]
[로직]
1. 커맨드 순회
    1-1. 커맨드 명령 수행
    1-2. 오프닝 구간 자동 스킵
2. 현재시간 출력 

"""